src/app/dashboard/dashboard.component.spec.ts

```diff
@@ -21,90 +23,116 @@ describe('DashboardComponent', () => {
   let component: DashboardComponent;
   let fixture: ComponentFixture<DashboardComponent>;
 
-  beforeEach(async(() => {
+  const configureTestbed = (errorMode = false): void => {
     TestBed.configureTestingModule({
       declarations: [DashboardComponent, createMockPipe('translate')],
       providers: [
         { provide: AuthenticationService, useClass: MockAuthenticationService },
-        { provide: DatasetsService, useClass: MockDatasetsService },
-        { provide: WorkflowService, useClass: MockWorkflowService },
+        {
+          provide: DatasetsService,
+          useClass: errorMode ? MockDatasetsServiceErrors : MockDatasetsService
+        },
+        {
+          provide: WorkflowService,
+          useClass: errorMode ? MockWorkflowServiceErrors : MockWorkflowService
+        },
         { provide: ErrorService, useClass: MockErrorService }
       ],
       schemas: [NO_ERRORS_SCHEMA]
     }).compileComponents();
-  }));
+  };
 
-  beforeEach(() => {
+  const b4Each = (): void => {
```

Comment:
The normal operation tests and the error mode tests each require (identical) functionality to be executed before each test.  Rather than duplicate identical functionality it was moved to this function, the alias for b4Each used because "beforeEach" already has meaning within the context.